src/Opaleye/Internal/Sql.hs

```diff
@@ -108,6 +118,7 @@ baseTable ti columns = SelectFrom $
               , tables = [Table (HSql.SqlTable (PQ.tiSchemaName ti) (PQ.tiTableName ti))] }
 
 product :: NEL.NonEmpty Select -> [HPQ.PrimExpr] -> Select
+product (t NEL.:| []) [] = t
```

Comment:
OK, so you're replacing a product of one select by the select itself.

1. Is this necessary to get your patch to work?  If so, why?  If not, let's do this in a different patch.

2. It should probably be done in (or alongside) [`mergeProduct`](https://github.com/ocharles/haskell-opaleye/blob/b070bc23d923097d4cf21d764339e09f846dcb53/src/Opaleye/Internal/Optimize.hs#L25) rather than here